integration_testing/connection_controller_routines_test.go

```diff
@@ -339,26 +414,121 @@ func TestConnectionControllerNonValidatorConnector(t *testing.T) {
+	require.NoError(err)
+	node5 := spawnValidatorNodeProtocol2(t, 18004, "node5", blsPriv5)
+	node5.Config.GlogV = 0
+	blsPriv6, err := bls.NewPrivateKey()
+	require.NoError(err)
+	node6 := spawnValidatorNodeProtocol2(t, 18005, "node6", blsPriv6)
+	node6.Config.GlogV = 0
+
+	node7 := spawnNonValidatorNodeProtocol2(t, 18006, "node7")
+	node8 := spawnNonValidatorNodeProtocol2(t, 18007, "node8")
+	node9 := spawnNonValidatorNodeProtocol2(t, 18008, "node9")
+	node10 := spawnNonValidatorNodeProtocol2(t, 18009, "node10")
 
 	node1 = startNode(t, node1)
-	node2 = startNode(t, node2)
 	defer node1.Stop()
+	node2 = startNode(t, node2)
 	defer node2.Stop()
+	node3 = startNode(t, node3)
+	defer node3.Stop()
+	node4 = startNode(t, node4)
+	defer node4.Stop()
+	node5 = startNode(t, node5)
+	defer node5.Stop()
+	node6 = startNode(t, node6)
+	defer node6.Stop()
+	node7 = startNode(t, node7)
+	defer node7.Stop()
+	node8 = startNode(t, node8)
+	defer node8.Stop()
+	node9 = startNode(t, node9)
+	defer node9.Stop()
+	node10 = startNode(t, node10)
+	defer node10.Stop()
+
+	// Connect node1 to node2, node3, node7, and node8.
+	cc1 := node1.Server.GetConnectionController()
+	require.NoError(cc1.CreateNonValidatorOutboundConnection(node2.Listeners[0].Addr().String()))
+	require.NoError(cc1.CreateNonValidatorOutboundConnection(node3.Listeners[0].Addr().String()))
+	require.NoError(cc1.CreateNonValidatorOutboundConnection(node7.Listeners[0].Addr().String()))
+	require.NoError(cc1.CreateNonValidatorOutboundConnection(node8.Listeners[0].Addr().String()))
+	// Connect node4, node5, node6 to node1.
+	cc4 := node4.Server.GetConnectionController()
+	require.NoError(cc4.CreateNonValidatorOutboundConnection(node1.Listeners[0].Addr().String()))
+	cc5 := node5.Server.GetConnectionController()
+	require.NoError(cc5.CreateNonValidatorOutboundConnection(node1.Listeners[0].Addr().String()))
+	cc6 := node6.Server.GetConnectionController()
+	require.NoError(cc6.CreateNonValidatorOutboundConnection(node1.Listeners[0].Addr().String()))
+
+	// Connect node9, node10 to node1.
+	cc9 := node9.Server.GetConnectionController()
+	require.NoError(cc9.CreateNonValidatorOutboundConnection(node1.Listeners[0].Addr().String()))
+	cc10 := node10.Server.GetConnectionController()
+	require.NoError(cc10.CreateNonValidatorOutboundConnection(node1.Listeners[0].Addr().String()))
+
+	setGetActiveValidatorImplWithValidatorNodes(t, node1, node2, node3, node4, node5, node6)
+	waitForValidatorConnection(t, node1, node2)
+	waitForValidatorConnection(t, node1, node3)
+	waitForValidatorConnection(t, node1, node4)
+	waitForValidatorConnection(t, node1, node5)
+	waitForValidatorConnection(t, node1, node6)
+	waitForCountRemoteNodeIndexerHandshakeCompleted(t, node1, 6, 5, 0, 1)
+}
 
-	waitForCountRemoteNodeIndexer(t, node1, 2, 0, 1, 1)
-	waitForCountRemoteNodeIndexer(t, node2, 2, 0, 1, 1)
+func TestConnectionControllerNonValidatorConnectorAddressMgr(t *testing.T) {
+	require := require.New(t)
```

Comment:
let's stop doing this? shadowing the package name is bad practice. If you want to create a new `*Assertions`, please save to a different variable name so it does not shadow the package name `require`. Please apply this to all tests where you do this.

Reply:
Are you sure? I think it's fineee. We've been doing this everywhere in our unit tests.

Reply:
I thought about it, and I agree with you. It's a bad practice. It's fixed now.